test/ItspServices.pServer.ClientTest/ProtectedDataClientTests.cs

```diff
@@ -28,122 +33,137 @@ protected override Task<HttpResponseMessage> SendAsync(HttpRequestMessage reques
         #endregion
 
         [TestMethod]
-        public async Task CreateProtectedData_ShouldSucceed()
+        public async Task CreateProtectedData_ShouldSendEncryptedDataAndKeyPair()
         {
-            bool setHasBeenCalled = false;
-
-            Mock<IHttpClientFactory> clientFactory = new Mock<IHttpClientFactory>();
-            HttpResponseMessage Callback(HttpRequestMessage request)
+            Mock<IApiClient> restClient = new Mock<IApiClient>();
+            Mock<ILocalKeysController> localKeysController = new Mock<ILocalKeysController>();
+            Mock<IDataEncryptor> dataEncryptor = new Mock<IDataEncryptor>();
+            List<bool> wasCalled = new List<bool>();
+            async Task<DataModel> CheckRequestDataByPath()
             {
-                string content = null;
-                switch (request.RequestUri.AbsolutePath)
-                {
-                    case "/api/protecteddata/folder/":
-                        content = "{\"ParentId\":null,\"Name\":\"root\",\"ProtectedDataIds\":[],\"SubfolderIds\":[1,2,3]}";
-                        break;
-                    case "/api/protecteddata/folder/1":
-                        content = "{\"ParentId\":null,\"Name\":\"FirstFolder\",\"ProtectedDataIds\":[],\"SubfolderIds\":[]}";
-                        break;
-                    case "/api/protecteddata/folder/2":
-                        content = "{\"ParentId\":null,\"Name\":\"Andys Passwords\",\"ProtectedDataIds\":[],\"SubfolderIds\":[4,5,6]}";
-                        break;
-                    case "/api/protecteddata/folder/3":
-                    case "/api/protecteddata/folder/4":
-                    case "/api/protecteddata/folder/5":
-                    case "/api/protecteddata/folder/6":
-                        Assert.Fail($"Client doesn't need to ask for this folder: {request.RequestUri.AbsolutePath}");
-                        break;
-                    case var s when s == "/api/protecteddata/data/1" && request.Method == HttpMethod.Get:
-                        content = "{\"Name\":\"MailAccount\",\"Data\":\"SecretPassword\"}";
-                        break;
-                    case var s when s == "/api/protecteddata/data/2" && request.Method == HttpMethod.Post:
-                        setHasBeenCalled = true;
-                        return new HttpResponseMessage(System.Net.HttpStatusCode.Created);
-                    default:
-                        Assert.Fail("Invalid requested URI");
-                        break;
-                }
-
-                return new HttpResponseMessage(System.Net.HttpStatusCode.OK)
-                {
-                    Content = new StringContent(content)
-                };
+                wasCalled.Add(true);
+                return null;
+            }
+            async Task<int> CheckSendCreateData()
+            {
+                wasCalled.Add(true);
+                return 1;
```

Comment:
To remove the CS1998 warning here you can remove the async keyword and change the return statement to
`return Task.FromResult(1);`